src/bundles/files/actions.js

```diff
@@ -398,6 +399,47 @@ const actions = () => ({
     }
   }),
 
+  /**
+   * Adds CAR file. On completion will trigger `doFilesFetch` to update the state.
+   * @param {FileStream} carFile
+   * @param {string} name
+   */
+  doAddCarFile: (carFile, name = '') => perform(ACTIONS.ADD_CAR_FILE, async (ipfs, { store }) => {
+    ensureMFS(store)
+
+    const stream = carFile.content.stream().pipeThrough(new TextDecoderStream())
+    const reader = stream.getReader()
+
+    let fileResultStr = ''
+    while (true) {
+      const { done, value } = await reader.read()
+      if (done) break
+      fileResultStr += value
+    }
+    const encoder = new TextEncoder()
+    const fileResult = encoder.encode(fileResultStr)
+    try {
+      const result = await all(ipfs.dag.import([fileResult], {
+        pinRoots: true
+      }))
```

Comment:
I think the problem here might be how we're reading the actual file bytes and passing it to dag.import.

We want something like https://github.com/ipfs/helia/blob/f7155d8bece43dd91d19060881294f61df9d222c/packages/interop/src/car.spec.ts#L60-L66

Basically we need to get the bytes of the carFile we get (which should be of type `File`) to bytes, and then pass those bytes (without encoding/decoding) to ipfs.dag.import.

An example of a car import process directly into the helia blockstore can be seen at https://github.com/ipfs/ipld-explorer-components/blob/57d625dadb934556506030fc532bf5ad24f36dc6/src/lib/import-car.ts#L12